src/haz3lcore/statics/Self.re

```diff
@@ -104,3 +104,9 @@ let list_concat = (ctx: Ctx.t, tys: list(Typ.t), ids: list(Id.t)): t =>
   | None => NoJoin(ty => List(ty), add_source(ids, tys))
   | Some(ty) => Just(ty)
   };
+
+let of_ann = (ctx: Ctx.t, ~p: Typ.t, ~ann: Typ.t) =>
+  switch (Typ.join_fix(ctx, ann, p)) {
```

Comment:
looking at the definition of `join_fix` more closely (which was from the previous PR), I'm confused -- it seems to be defined asymmetrically, with Unknown(TypHole) on the right behaving differently than Unknown(TypHole) on the left. I would expect a join to be symmetric. That asymmetry seems to be load-bearing here in favoring the direct annotation on the variable